### PR TITLE
1731 Add TrackingEventsHandler to support Pulse SDK. Track errors and interactions for Simplified Login 

### PR DESCRIPTION
**Pull request by zamorastp**

### CODE REVIEW

File: Sources/AccountSDKIOSWeb/Client/Client.swift [hunk 1]
```diff
@@ -168,14 +186,19 @@ public class Client: CustomStringConvertible {
...
         let session = ASWebAuthenticationSession(url: url, callbackURLScheme: clientScheme) { callbackURL, error in
             guard let url = callbackURL else {
                 if case ASWebAuthenticationSessionError.canceledLogin? = error {
                     SchibstedAccountLogger.instance.debug("Login flow was cancelled")
+                    self.tracker?.engagement(.click(on: .cancel), in: .webBrowser)
                     completion(.failure(.canceled))
                 } else {
                     SchibstedAccountLogger.instance.error("Login flow error: \(String(describing: error))")
-                    completion(.failure(.unexpectedError(message: "ASWebAuthenticationSession failed: \(String(describing: error))")))
+                    let error = LoginError.unexpectedError(message: "ASWebAuthenticationSession failed: \(String(describing: error))")
```
Comment (clausjoergensen) [inline Sources/AccountSDKIOSWeb/Client/Client.swift:199]: `String(describing: error)` will always print `Optional<String>(value)` even if it's not nill. Let's please not have that in the analytics. 

Instead you should unwrap the error in the `else`

File: Sources/AccountSDKIOSWeb/Client/Client.swift [hunk 2]
```diff
@@ -168,14 +186,19 @@ public class Client: CustomStringConvertible {
...
         
+        tracker?.interaction(.view, with: .webBrowser, additionalFields: [.getLoginSession(withMFA), .loginHint(loginHint), .withAssertion((assertion != nil) ? true : false), .extraScopeValues(extraScopeValues)])
+        
         let session = ASWebAuthenticationSession(url: url, callbackURLScheme: clientScheme) { callbackURL, error in
             guard let url = callbackURL else {
                 if case ASWebAuthenticationSessionError.canceledLogin? = error {
                     SchibstedAccountLogger.instance.debug("Login flow was cancelled")
+                    self.tracker?.engagement(.click(on: .cancel), in: .webBrowser)
                     completion(.failure(.canceled))
                 } else {
```
Comment (clausjoergensen) [inline Sources/AccountSDKIOSWeb/Client/Client.swift:197]:
```suggestion
                } else if let error = error {
```

File: Sources/AccountSDKIOSWeb/Client/Client.swift [hunk 3]
```diff
@@ -168,14 +186,19 @@ public class Client: CustomStringConvertible {
             preconditionFailure("Couldn't create loginURL")
         }
         
+        tracker?.interaction(.view, with: .webBrowser, additionalFields: [.getLoginSession(withMFA), .loginHint(loginHint), .withAssertion((assertion != nil) ? true : false), .extraScopeValues(extraScopeValues)])
```
Comment (clausjoergensen) [inline Sources/AccountSDKIOSWeb/Client/Client.swift:189]: nit: very long line, can we break it up?

File: Sources/AccountSDKIOSWeb/Client/Client.swift [hunk 4]
```diff
@@ -117,17 +131,20 @@ public class Client: CustomStringConvertible {
                   stateStorage: stateStorage,
                   httpClient: chttpClient,
                   jwks: jwks,
-                  tokenHandler: tokenHandler)
+                  tokenHandler: tokenHandler,
+                  tracker: tracker)
     }
     
-    init(configuration: ClientConfiguration, sessionStorage: SessionStorage, stateStorage: StateStorage, httpClient: HTTPClient, jwks: JWKS, tokenHandler: TokenHandler) {
+    init(configuration: ClientConfiguration, sessionStorage: SessionStorage, stateStorage: StateStorage, httpClient: HTTPClient, jwks: JWKS, tokenHandler: TokenHandler, tracker: TrackingEventsHandler? = nil) {
```
Comment (clausjoergensen) [inline Sources/AccountSDKIOSWeb/Client/Client.swift:138]: nit: very long line, can you break it down like you did above?

File: Sources/AccountSDKIOSWeb/Lib/MFAType.swift [hunk 5]
```diff
@@ -13,7 +13,7 @@ public enum MFAType: Codable {
     /// BankId verification for PRE environment
     case preEid(PreEidType)
 
-  var rawValue: String {
+  public var rawValue: String {
```
Comment (clausjoergensen) [inline Sources/AccountSDKIOSWeb/Lib/MFAType.swift:16]: Why was this made public?

File: Sources/AccountSDKIOSWeb/Client/Client.swift (hunk 4 above)
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Client/Client.swift]: Good point!

(Hope there will be a time in the future to add SwiftLint or swift-format to the project)

File: Sources/AccountSDKIOSWeb/Client/Client.swift (hunk 1 above)
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Client/Client.swift]: Done

File: Sources/AccountSDKIOSWeb/Client/Client.swift (hunk 2 above)
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Client/Client.swift]: Done

File: Sources/AccountSDKIOSWeb/Client/Client.swift (hunk 3 above)
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Client/Client.swift]: Done

File: Sources/AccountSDKIOSWeb/Lib/MFAType.swift (hunk 5 above)
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Lib/MFAType.swift:16]: I use it to set acr value in additional fields (for pre env there is another type PreEidType). Let's try if I can do it without making it explicitly public

**Review by clausjoergensen — APPROVED** [review]

**Pull request merged**